app/backend/src/couchers/models.py

```diff
@@ -867,26 +875,35 @@ def __repr__(self):
         return f"Message(id={self.id}, time={self.time}, text={self.text}, author={self.author}, conversation={self.conversation})"
 
 
-class Complaint(Base):
+class Report(Base):
     """
-    A record that a user has reported another user to admin
+    A record that a user has reported content to admin
     """
 
-    __tablename__ = "complaints"
+    __tablename__ = "reports"
 
     id = Column(BigInteger, primary_key=True)
 
-    # timezone should always be UTC
     time = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
 
-    author_user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
-    reported_user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
+    # the user who reported or flagged
+    reporting_user_id = Column(ForeignKey("users.id"), nullable=False, index=True)
 
+    # reason, e.g. spam, inappropriate, etc
     reason = Column(String, nullable=False)
+    # a short description
     description = Column(String, nullable=False)
 
-    author_user = relationship("User", foreign_keys="Complaint.author_user_id")
-    reported_user = relationship("User", foreign_keys="Complaint.reported_user_id")
+    # a reference to the content
+    content_ref = Column(String, nullable=False)
+    # the author of the content
+    author_user_id = Column(ForeignKey("users.id"), nullable=False)
+
+    user_agent = Column(String, nullable=False)
+    page = Column(String, nullable=False)
+
+    reporting_user = relationship("User", foreign_keys="Report.reporting_user_id")
+    author_user = relationship("User", foreign_keys="Report.author_user_id")
```

Comment:
Aren't the reporting user also the author?

Reply:
The purpose is for `reporting_user` to be the user that pressed the flag button to report the content to moderators/support.

`author_user` is the user who authored and created that object.

E.g. if there is a spam comment, the person who wrote the comment is the author, the person who clicked the button is the reporting user.

I don't like the confusion here either, but I'm having a hard time coming up with better names?

Reply:
Aha! I parsed "author" as the author of the report. I can't think about any better name, a comment would be good to clarify it is the author of the offending content.

Reply:
Will do!